### PR TITLE
Shim funded channels: don't count towards max pending channels, allow to abandon them

### PR DESCRIPTION
**Pull request by guggero**
Externally funded channels are expected by the user and explicitly
registered through the use of a funding shim and should therefore not
count towards the max pending channel count which is primarily there to
mitigate DoS attacks.
For channels that are created through a shim but never confirm, we want to allow them to be removed through the RPC without the need for a dev flag.

### CODE REVIEW

File: fundingmanager.go [hunk 1]
```diff
@@ -1221,7 +1233,13 @@ func (f *fundingManager) handleFundingOpen(fmsg *fundingOpenMsg) {
 	}
 
 	for _, c := range channels {
-		if c.IsPending {
+		// Pending channels that have a non-zero thaw height were also
+		// created through a canned funding shim. Those also don't
+		// count towards the DoS protection limit.
+		//
+		// TODO(guggero): Properly store the funding type (wallet, shim,
```
Comment (halseth) [inline fundingmanager.go:1240]: to resolve this TODO, would we need db change?

File: lnrpc/rpc.proto [hunk 2]
```diff
@@ -3052,11 +3054,20 @@ message DeleteAllPaymentsResponse {
...
 message AbandonChannelRequest {
     ChannelPoint channel_point = 1;
+
+    bool pending_funding_shim_only = 2;
 }
 
 message AbandonChannelResponse {
 }
 
+message ForgetPendingExternalChannelRequest {
```
Comment (halseth) [inline lnrpc/rpc.proto:3064]: not used?

**Review by halseth — CHANGES_REQUESTED** [review]

File: fundingmanager.go (hunk 1 above)
Comment (guggero) [reply fundingmanager.go:1240]: Yes. Though with TLV not necessarily a migration. Would like to get #4517 in first that introduces the TLV for channel serialization.

File: lnrpc/rpc.proto (hunk 2 above)
Comment (guggero) [reply lnrpc/rpc.proto]: Oops, removed.

**Comment by guggero** [comment]
Fixed and rebased.

**Review by halseth — APPROVED** [review]
LGTM 🏆

**Pull request merged**